### PR TITLE
mimic: mgr: Release GIL and Balancer fixes

### PR DESCRIPTION
**Pull request by dzafman**
Backports:
#31064
#31542
#31774
#31990
#32247

Parent trackers
https://tracker.ceph.com/issues/42432
https://tracker.ceph.com/issues/42756
https://tracker.ceph.com/issues/42718
https://tracker.ceph.com/issues/43084 - no backport tracker
https://tracker.ceph.com/issues/43312

Backport trackers
https://tracker.ceph.com/issues/42697
https://tracker.ceph.com/issues/42798
https://tracker.ceph.com/issues/43094
https://tracker.ceph.com/issues/43530

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by smithfarm** [comment]
@dzafman Does this need https://github.com/ceph/ceph/pull/31944 as well?

**Comment by dzafman** [comment]
Testing passed
http://pulpito.ceph.com/dzafman-2019-12-05_20:01:02-rados-wip-zafman-testing-mimic-distro-basic-smithi/

**Comment by dzafman** [comment]
jenkins test make check

**Comment by smithfarm** [comment]
Please don't merge this until v13.2.8 is safely out.

**Comment by dzafman** [comment]
@smithfarm Should I add 13.2.8 in the PendingReleaseNotes here?

**Comment by dzafman** [comment]
Waiting for #32247 to merge and then added here before merging

File: PendingReleaseNotes [hunk 1]
```diff
@@ -45,3 +45,13 @@
   value determined by the config options, for the average for any of the 3 intervals.
   New admin command ``ceph daemon osd.# dump_osd_network [threshold]`` will
   do the same but only including heartbeats initiated by the specified OSD.
+
+* The configuration value ``osd_calc_pg_upmaps_max_stddev`` used for upmap
+  balancing has been removed. Instead use the mgr balancer config
+  ``upmap_max_deviation`` which now is an integer number of PGs of deviation
+  from the target PGs per OSD.  This can be set with a command like
+  ``ceph config set mgr mgr/balancer/upmap_max_deviation 2``.  The default
+  ``upmap_max_deviation`` is 1.  There are situations where crush rules
```
Comment (neha-ojha) [inline PendingReleaseNotes]: We should update this.

**Comment by yuriw** [comment]
[wip-yuri5-testing-2020-01-16-1519-mimic](https://trello.com/c/KiiWY8BZ/809-wip-yuri5-testing-2020-01-16-1519-mimic)

**Comment by yuriw** [comment]
retest this please

**Pull request merged**